### PR TITLE
feature-136-storeknotcoordinates

### LINKED ISSUE #138: Knot coordinates
As a developer I would like to store created and updated knot coordinates

### PR DESCRIPTION
**Pull request by Daniilmass**
resolves #138

### CODE REVIEW

File: src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java [hunk 1]
```diff
@@ -174,13 +174,15 @@ public Response createIntent(Intent body, UUID storyId) {
   @Override
   public Response createKnot(Knot body, UUID storyId) {
     UUID loggedUserId = getLoggerUserId();
+    Double coordinateX = body.getCoordinates().getX();
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java]: possible null pointer if coordinates is null

File: src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java [hunk 2]
```diff
@@ -556,6 +558,8 @@ public Response updateIntent(Intent body, UUID storyId, UUID intentId) {
   @Override
   public Response updateKnot(Knot body, UUID storyId, UUID knotId) {
     UUID loggedUserId = getLoggerUserId();
+    Double coordinateX = body.getCoordinates().getX();
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/server/rest/StoriesApiImpl.java]: possible null pointer if coordinates is null

File: src/main/java/fi/metatavu/metamind/server/rest/translation/KnotTranslator.java [hunk 3]
```diff
@@ -33,6 +35,8 @@ public Knot translateKnot(fi.metatavu.metamind.persistence.models.Knot jpaKnot)
     result.setName(jpaKnot.getName());
     result.setStoryId(jpaKnot.getStory() != null ? jpaKnot.getStory().getId() : null);
     result.setType(jpaKnot.getType());
+    result.setCoordinates(coordinates.x(jpaKnot.getCoordinateX()));
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/server/rest/translation/KnotTranslator.java]: this is bit confusing, set coordinate x and y and then set coordinates to result only once (coordinates.setX(x), coordinates.setY(y) .... result.setCoordinates(coordinates))

File: src/test/java/fi/metatavu/metamind/test/functional/KnotTestsIT.java [hunk 4]
```diff
@@ -107,7 +113,7 @@ public void testDeleteKnot() throws Exception {
   public void testDeleteKnotPermissions() throws Exception {
     try (TestBuilder builder = new TestBuilder()) {
       Story story = builder.admin().stories().create("en", "test story", "Enter your answer");
-      Knot createdKnot = builder.admin().knots().create(story, KnotType.TEXT, "Test", "Content");
+      Knot createdKnot = builder.admin().knots().create(story, KnotType.TEXT, "Test", "Content", 10.0, 20.0);
       builder.anonymous().knots().assertDeleteFailStatus(story, 401, createdKnot);
       builder.invalid().knots().assertDeleteFailStatus(story, 403, createdKnot);
     }
```
Comment (hkurhinen) [inline src/test/java/fi/metatavu/metamind/test/functional/KnotTestsIT.java]: Add a test that tests also null coordinates

File: src/test/java/fi/metatavu/metamind/test/functional/builder/impl/KnotTestBuilderResource.java [hunk 5]
```diff
@@ -31,15 +32,20 @@ public KnotTestBuilderResource(TestBuilder testBuilder, ApiClient apiClient) {
...
-  public Knot create(Story story, KnotType type, String name, String content) {
+  public Knot create(Story story, KnotType type, String name, String content, Double coordinateX, Double coordinateY) {
     Knot knot = new Knot();
+    Coordinates coordinates = new Coordinates();
     knot.setType(type);
     knot.setName(name);
     knot.setContent(content);
     knot.setTokenizer(TokenizerType.WHITESPACE);
     knot.setHint("hint");
+    knot.setCoordinates(coordinates.x(coordinateX));
```
Comment (hkurhinen) [inline src/test/java/fi/metatavu/metamind/test/functional/builder/impl/KnotTestBuilderResource.java]: set the x and y before setting coordinates to knot.
